### ISSUE TITLE
[Experimental] Looking for testers of new 0.14.51.rc3.3 syncthing version

### ISSUE THREAD

**Issue by Catfriend1** [Catfriend1/syncthing-android#44]
Hi,

I'm currently testing syncthing (core) 0.14.51-rc.3 from upstream repository if it works well in the android wrapper. As event handling for example changed upstream, I need to test this before releasing a new version - and if there are problems - adjust the wrapper code.

I appreciate your help on this. If you'd like to help, **please backup any important data and your syncthing configuration** - then report any issues found during testing this APK referring to its version and date.

**Comment by Catfriend1** [comment]
First glance: Looks good :)
![image](https://user-images.githubusercontent.com/16361913/45406355-7e7f6e00-b666-11e8-9016-a5ee4e5a522a.png)

**Comment by Catfriend1** [comment]
@dkaparis @Avamander : Hi you two. Would you mind helping me on testing the most recent syncthing version and report back if your issues (reported on the official issue tracker) still exist?

**Comment by dkaparis** [comment]
I would, but schedule is very tight these days. Maybe I'll get around to it end of next week or so.

**Comment by Catfriend1** [comment]
@dkaparis no hurry :)

**Comment by Avamander** [comment]
@Catfriend1  the app crashes on the key generation screen.

```Java
E/AndroidRuntime( 9864): FATAL EXCEPTION: AsyncTask #1
E/AndroidRuntime( 9864): Process: com.github.catfriend1.syncthingandroid.debug, PID: 9864
E/AndroidRuntime( 9864): java.lang.RuntimeException: An error occured while executing doInBackground()
E/AndroidRuntime( 9864):        at android.os.AsyncTask$3.done(AsyncTask.java:300)
E/AndroidRuntime( 9864):        at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:355)
E/AndroidRuntime( 9864):        at java.util.concurrent.FutureTask.setException(FutureTask.java:222)
E/AndroidRuntime( 9864):        at java.util.concurrent.FutureTask.run(FutureTask.java:242)
E/AndroidRuntime( 9864):        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
E/AndroidRuntime( 9864):        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
E/AndroidRuntime( 9864):        at java.lang.Thread.run(Thread.java:841)
E/AndroidRuntime( 9864): Caused by: android.view.ViewRootImpl$CalledFromWrongThreadException: Only the original thread that created a view hierarchy can touch its views.
E/AndroidRuntime( 9864):        at android.view.ViewRootImpl.checkThread(ViewRootImpl.java:6027)
E/AndroidRuntime( 9864):        at android.view.ViewRootImpl.requestLayout(ViewRootImpl.java:820)
E/AndroidRuntime( 9864):        at android.view.View.requestLayout(View.java:16455)
E/AndroidRuntime( 9864):        at android.view.View.requestLayout(View.java:16455)
E/AndroidRuntime( 9864):        at android.view.View.requestLayout(View.java:16455)
E/AndroidRuntime( 9864):        at android.view.View.requestLayout(View.java:16455)
E/AndroidRuntime( 9864):        at android.widget.RelativeLayout.requestLayout(RelativeLayout.java:352)
E/AndroidRuntime( 9864):        at android.view.View.requestLayout(View.java:16455)
E/AndroidRuntime( 9864):        at android.view.View.requestLayout(View.java:16455)
E/AndroidRuntime( 9864):        at android.widget.RelativeLayout.requestLayout(RelativeLayout.java:352)
E/AndroidRuntime( 9864):        at android.view.View.requestLayout(View.java:16455)
E/AndroidRuntime( 9864):        at android.view.View.requestLayout(View.java:16455)
E/AndroidRuntime( 9864):        at android.widget.TextView.checkForRelayout(TextView.java:6600)
E/AndroidRuntime( 9864):        at android.widget.TextView.setText(TextView.java:3813)
E/AndroidRuntime( 9864):        at android.widget.TextView.setText(TextView.java:3671)
E/AndroidRuntime( 9864):        at android.widget.TextView.setText(TextView.java:3646)
E/AndroidRuntime( 9864):        at com.nutomic.syncthingandroid.activities.FirstStartActivity$KeyGenerationTask.doInBackground(FirstStartActivity.java:484)
E/AndroidRuntime( 9864):        at com.nutomic.syncthingandroid.activities.FirstStartActivity$KeyGenerationTask.doInBackground(FirstStartActivity.java:465)
E/AndroidRuntime( 9864):        at android.os.AsyncTask$2.call(AsyncTask.java:288)
E/AndroidRuntime( 9864):        at java.util.concurrent.FutureTask.run(FutureTask.java:237)
E/AndroidRuntime( 9864):        ... 3 more
W/ActivityManager(  600):   Force finishing activity com.github.catfriend1.syncthingandroid.debug/com.nutomic.syncthingandroid.activities.FirstStartActivity
```

**Comment by Catfriend1** [comment]
@Avamander I'll look into that, thanks for reporting the problem.

**Comment by Catfriend1** [comment]
Wow, hard to keep up, but I'm preparing to test 0.14.51-rc.3 ...

**Comment by Catfriend1** [comment]
@Avamander Referring to #46 the current fdroid build is badly broken. I've fixed it and hoping fdroid will update its store soon.

**Comment by Catfriend1** [comment]
Bug #50 discovered related to most recent st release 0.14.51-rc.3

**Comment by Catfriend1** [comment]
@Avamander Which Android version do you use? Is that crash happening on Android 4 or 5?

**Comment by Catfriend1** [comment]
@dkaparis @Avamander I've made a new release with a lots of bugs and crashes fixed. The wrapper is now ready for the most recent syncthing 0.14.51-rc.3.
https://github.com/Catfriend1/syncthing-android/releases/tag/v0.14.51.rc3.3
https://github.com/Catfriend1/syncthing-android/releases/download/v0.14.51.rc3.3/com.github.catfriend1.syncthingandroid_4159_0.14.51.rc3.3_861d92b.apk

**Comment by Catfriend1** [comment]
We are now live on fdroid, you can also get the app there if you prefer.

**Comment by Avamander** [comment]
![screenshot_2018-09-19-00-59-11](https://user-images.githubusercontent.com/5392141/45719067-d9a1eb00-bba7-11e8-890c-68be1a6c9c77.png)


a. The setup text overflows the bottom bar
b. The `Continue` button text gets clipped by physical screen borders
c. When key generation fails the back button takes one back but then there's no `Continue` button any longer and the app has to be restarted - just a bit confusing UI
d. The key generation fails with this stacktrace:
```
java.io.IOException: Error running exec(). Command: [/data/app-lib/com.github.catfriend1.syncthingandroid.debug-2/libsyncthing.so, -generate, /data/data/com.github.catfriend1.syncthingandroid.debug/files, -logflags=0] Working Directory: null Environment: [STTRACE=, ANDROID_ROOT=/system, ANDROID_CACHE=/cache, LOOP_MOUNTPOINT=/mnt/obb, ANDROID_BOOTLOGO=1, STNORESTART=1, STHASHING=minio, STGUIASSETS=/storage/sdcard0/Android/data/com.github.catfriend1.syncthingandroid.debug/files/gui, TERMINFO=/system/etc/terminfo, LD_LIBRARY_PATH=/vendor/lib:/system/lib, EXTERNAL_STORAGE=/storage/sdcard0, ANDROID_SOCKET_zygote=10, ANDROID_DATA=/data, PATH=/sbin:/vendor/bin:/system/sbin:/system/bin:/system/xbin, ANDROID_ASSETS=/system/app, ASEC_MOUNTPOINT=/mnt/asec, HOME=/storage/sdcard0, BOOTCLASSPATH=/system/framework/core.jar:/system/framework/conscrypt.jar:/system/framework/okhttp.jar:/system/framework/core-junit.jar:/system/framework/bouncycastle.jar:/system/framework/ext.jar:/system/framework/framework.jar:/system/framework/framework2.jar:/system/framework/telephony-common.jar:/system/framework/voip-common.jar:/system/framework/mms-common.jar:/system/framework/android.policy.jar:/system/framework/services.jar:/system/framework/apache-xml.jar:/system/framework/webviewchromium.jar:/system/framework/telephony-msim.jar, ANDROID_PROPERTY_WORKSPACE=8,0, STNOUPGRADE=1, SECONDARY_STORAGE=/storage/usbdisk0, ANDROID_STORAGE=/storage]
E/SyncthingRunnable(32172):     at java.lang.ProcessManager.exec(ProcessManager.java:211)
E/SyncthingRunnable(32172):     at java.lang.ProcessBuilder.start(ProcessBuilder.java:195)
E/SyncthingRunnable(32172):     at com.nutomic.syncthingandroid.service.SyncthingRunnable.setupAndLaunch(SyncthingRunnable.java:508)
E/SyncthingRunnable(32172):     at com.nutomic.syncthingandroid.service.SyncthingRunnable.run(SyncthingRunnable.java:145)
E/SyncthingRunnable(32172):     at com.nutomic.syncthingandroid.service.SyncthingRunnable.run(SyncthingRunnable.java:107)
E/SyncthingRunnable(32172):     at com.nutomic.syncthingandroid.util.ConfigXml.<init>(ConfigXml.java:68)
E/SyncthingRunnable(32172):     at com.nutomic.syncthingandroid.activities.FirstStartActivity$KeyGenerationTask.doInBackground(FirstStartActivity.java:481)
E/SyncthingRunnable(32172):     at com.nutomic.syncthingandroid.activities.FirstStartActivity$KeyGenerationTask.doInBackground(FirstStartActivity.java:465)
E/SyncthingRunnable(32172):     at android.os.AsyncTask$2.call(AsyncTask.java:288)
E/SyncthingRunnable(32172):     at java.util.concurrent.FutureTask.run(FutureTask.java:237)
E/SyncthingRunnable(32172):     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
E/SyncthingRunnable(32172):     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
E/SyncthingRunnable(32172):     at java.lang.Thread.run(Thread.java:841)
E/SyncthingRunnable(32172): Caused by: java.io.IOException: No such file or directory
E/SyncthingRunnable(32172):     at java.lang.ProcessManager.exec(Native Method)
E/SyncthingRunnable(32172):     at java.lang.ProcessManager.exec(ProcessManager.java:209)
E/SyncthingRunnable(32172):     ... 12 more
```

Thank you for your effort at fixing these bugs!

**Comment by Catfriend1** [comment]
Thanks for your feedback. From the log above I see that the Syncthing binary is not there. Honestly I dont know why , is this an old android version, if yes , which one?

**Comment by Catfriend1** [comment]
Hi @rominail ,
Would you mind helping us to test the recent release of syncthing fork, receive folder support is already there.
Kind regards 
Catfriend1

**Comment by Catfriend1** [comment]
Hi @Avamander 
according to PR #67 I've fixed the bugs you've mentioned. But as you've gotten hit by a fundamental problem where libSyncthing.so cannot be found in the expected location I cannot fix the root cause for you as I suspect it's a problem with your specific android version.
The app will no longer request root if you did not enable the preference checkbox in the settings to do so.
If the syncthing core binary cannot be executed because it's missing, you'll get a clear notice and path information on the welcome wizard (but still, it's the end of story then without any chance to hit "next").
Feel free to try it and post a screenshot of the failure again so I have more detailed debug info on this than in the last log.
Kind regards

**Comment by Catfriend1** [comment]
@leewdch A newer build is on this report providing a status tab to diagnose why synching is or is not running. Feel free to try it and if it doesn't work, open an issue here.

**Comment by Catfriend1** [comment]
@jessse4216 Does the current release from here fix the WiFi run condition issue?

**Comment by jessse4216** [comment]
@catfriend1 hi thanks, I tried both versions v0.14.51.rc4  and 0.14.51-rc.3 from this thread, still same problem. I select my home network that I am current on, then syncthing becomes disabled.

**Comment by jessse4216** [comment]
@Catfriend1 yes, that's it.  I had location turned off from the pull down drawer at the top.  Once I turned on, it worked.  I usually don't have location on however due to battery and privacy concerns. I do have the location permission on for syncthing on however.   Is there a way to work around it? I know in the app you have to use location services in order to get the wifi SSID.

**Comment by leewdch** [comment]
I just found out I need to manually enable location in my privacy settings for this to work. Only granting locaiton permissions to syncthing without enabling location doesn't work. I also set location to work only on wifi and mobile data to save battery and it seems to work. I'll try now to exit the syncthing fork and repeat on the stable app to see if this solves the issue on stable as well

**Comment by Catfriend1** [comment]
@jessse4216 I've already looked for a workaround some weeks ago but didn't find a valid implementation. E.g. shell reading of the WiFi drivers assoc ssid is also blocked by Android when location is off. It seems to be a hard system restriction.

**Comment by jessse4216** [comment]
@Catfriend1 Thanks again. Is there another way to force syncthing to only send data on a local network? That is all I am really worried about in setting syncthing to just use my home wifi.

**Comment by leewdch** [comment]
@jessse4216 you can manually disable location for all the apps but syncthing in `Settings > Apps & notifications > App permissions > Location` and set location to work on wifi/mobile data only in `Settings > Security & privacy > Location > Mode > Battery saving`

**Comment by jessse4216** [comment]
@Catfriend1 @leewdch Thank you both.

**Comment by Catfriend1** [comment]
@jessse4216 To restrict syncing data in the local network I suggest going to settings in the app> syncthing options > turn off relay, global discovery and nat traversal. So just local discovery is left on.

**Comment by jessse4216** [comment]
@Catfriend1 awesome thanks so much.

**Comment by Catfriend1** [comment]
@jessse4216 I've created issue #82 and will implement an in-app guide that helps users wanting the same like you to configure it correctly.

**Comment by Catfriend1** [comment]
@schmendrik Would you mind trying the forked version to make sure we don't talk about a problem I've already solved. Upstream is unfortunately far behind Android 8+ compatibility while the fork doesn't have enough users to prove my fixes working.

**Comment by schmendrik** [comment]
@Catfriend1 Of course, I'll test it. Do I need to backup my current settings, if so how? Don't want to re-setup everything for and after the testing.

**Comment by Catfriend1** [comment]
Yes, please backup under the preferences screen in the official syncthing app. This backup can be imported by pulling the drawer and selecting "import and export" in the forked version.

**Comment by schmendrik** [comment]
@Catfriend1 Your fork runs perfectly fine. My [issue](https://github.com/syncthing/syncthing-android/issues/1193#issuecomment-430051051) is gone. One thing though: 

![screenshot_2018-10-17-20-23-55](https://user-images.githubusercontent.com/15148632/47108698-9e4d1780-d24c-11e8-876a-de7f44c034d3.png)

It's been a long time since I've (re)-installed syncthing. As you can see, the text goes well below what it's supposed to. Also, Battery Optimization is one permission I did /not/ grant the official syncthing version on purpose. With this Welcome screen, I could not continue unless I granted that permission. 

Thank you, I'm going to keep this version for now. One other issue I've been having with syncthing is extreme battery drain when the app's running. Let's see if this version fixes that too.

**Comment by schmendrik** [comment]
Interestingly enough, with your version, I've only lost 1% battery so far, whereas the previous (official) version would have consumed about 30 to 50% by now. To be sure, I'll have to wait and observe over the course of a few weeks. There is one more stress test that I will be conducting in a few days.

**Comment by Catfriend1** [comment]
@schmendrik Thanks for reporting back. I made the "ignore battery optimizations" mandatory for technical reasons. This is a long story starting out with Android 6+ and requires reading a lot through the docs to understand how Google designed Android and changed behaviour across versions. It doesn't harm your battery granting it but if anything goes wrong within the syncthing binary your battery will drain - not only with screen on but also with screen off. My point of view is, wrapper and core binary should be working properly (or be fixed) and then the ignore battery optimizations are there to tell Android not to limit syncthings possibilities when the screen goes off.
The overflowing texts should be fixed, yes. I'm not a good designer but will soon get to this.

**Comment by Catfriend1** [comment]
@MartinX3 Regarding your problem with the battery eater, would you mind trying the forked version as I don't have Android 9 yet. Many root pop-ups look okay so far because the wrapper needs to use the Shell frequently. You can also write to sdcard without root but only to special paths, the forked version has a guide suggesting this special path while you are able to try other paths and the UI shows you which permission level Android os allows there.

**Comment by MartinX3** [comment]
@Catfriend1 
I installed it, made a new directory to share and choose the path on SDCard.
Sadly there was no guide and it is telling me, that it can't write.

**Comment by Catfriend1** [comment]
@MartinX3 okay. What you describe is the Android restriction. We cannot fix this, I'm sorry.

**Comment by Catfriend1** [comment]
It has technical reasons, go binaries cannot write as they cannot use the Java api Android enforced since 4.4

**Comment by MartinX3** [comment]
Thank you for this explanation. :)

So maybe you want to use a Abstraction Layer?
A Helper Routine in Java?

**Comment by Catfriend1** [comment]
Ahem.... Yes, but I don't consider this my task even though it my be possible and there is already an existing bounty for that. Many people previously discussing this already stated they consider this a LOT of work. You also need people on the other side (so syncthing Android and syncthing core) that work together for an interface between go and Java.

**Comment by MartinX3** [comment]
Yes of course, that's right.

On the other site slows the root this app and my system down, or the "root" message is the cause.
And more people will accept this app, because not every one is willing to give this app root privileges.

So I hope it may happen in the future. :)

**Comment by Catfriend1** [comment]
In the meantime, it always works syncing to /storage/xxxx-yyyy/Android/data/com.github.catfriend1.syncthingandroid/files as Android allows read write a ccess there. More about this is described in the wiki.

**Comment by MartinX3** [comment]
Yes, I forgot this aspect. Thank you! :)

**Comment by MartinX3** [comment]
Btw, I can't changed the standard folder path for new folders.
I changed it in the web interface, but it doesn't change in the app itself.

**Comment by Catfriend1** [comment]
We should file this as a separate issue and see if the default path can be made the same.

**Comment by schmendrik** [comment]
@Catfriend1 Are the custom sync conditions per folder not working? I can't activate any sync conditions other than "Run on WiFi". They are disabled/greyed out.

**Comment by Catfriend1** [comment]
@schmendrik The new per folder / per device run conditions still need to prove outside my test lab and own use cases, they are part of a possible future transition to "pause/unpause folders/devices" instead of "globally" starting and stopping syncthing. I agree the UI is not best now, it is planned to improve it at later milestone. I'm waiting for user feedback first regarding
a) battery usage (full stop and start vs. pause and unpause by custom conditions)
b) reliability and stability of the new implementation.
So yes, you can expect that this will get improved but not in this early stage. Another reason that made me prefer the current implementation was not to "totally screw existing configs and settings done by users" - users can upgrade to the recent release and if they don't enable the new settings synching-android just behaves like in earlier versions not breaking any concept the user might rely on the app does "what it should". With a good mind to improve things further, kind regards 
Catfriend1

**Comment by Catfriend1** [comment]
@Rakleed Hi, I've fixed the file manager stuff since the last two releases and also the folder creation stuff. Would you mind trying the forked version and feedback?

**Comment by Rakleed** [comment]
@Catfriend1 still the same mistake.
Log: [2018-10-23-23-33-56.txt](https://github.com/Catfriend1/syncthing-android/files/2507926/2018-10-23-23-33-56.txt)

**Comment by Catfriend1** [comment]
@Rakleed ok, I can take care of fixing this. What Android ROM , version and device manufacturer , model are we talking about here?

**Comment by Rakleed** [comment]
LineageOS 15.1, Android 8.1.0, OnePlus 3.

**Comment by Rakleed** [comment]
@Catfriend1 LineageOS 15.1, Android 8.1.0, OnePlus 3.

**Comment by Catfriend1** [comment]
@blaueente Would you mind to try the forked version and report back if your problem regarding the root processes is solved? https://github.com/Catfriend1/syncthing-android/releases

**Comment by blaueente** [comment]
Thanks for the hint! I will try this, but it might take some time. I'd prefer the F-droid version, if you don't mind (still v0.14.51.10 / 4173)

**Comment by Catfriend1** [comment]
@blaueente No problem, head to F-Droid and grab v0.14.51.10 as I didn't change anything related to your problem in 4174.

**Comment by Catfriend1** [comment]
@noodle- could you please try and confirm if the battery eater problem is solved in the forked version ? It's available here and on fdroid.

**Comment by Catfriend1** [comment]
Yes , it still applies, but it's no longer RC and can be considered stable: https://github.com/Catfriend1/syncthing-android/releases/tag/v0.14.51.13

**Comment by schmendrik** [comment]
I've updated to 0.14.51.13 a few days ago and haven't run into any issues (so much for my testing).

@Catfriend1, what is updating syncthing-android going to look like in the future? Will it be merged into offical syncthing-android at some point? I'm really thankful for your build by the way and wouldn't mind updating manually. I'm just glad I have a stable syncthing android app running on my phone right now.

**Comment by Catfriend1** [comment]
Hi @schmendrik ,
at the time of writing this I cannot tell where syncthing-android official is going to. I've noticed that my access to the gplay error report console was revoked last night, wrote to the core team what happened and that was justified because I quit the official repo in August. To not harden the fronts further, I contacted calmh who moved this discussion to private on the forum and I'm now seeking a solution for the "do we / how we work together" question with him. My main problem and opinion is that working on the official app is still possible but there would (currently) only be me as code contributor and a lot of review process I would have to take to get every single change in. I understand the teams position, especially what calmh wrote me in private that a github open source project is about collaborating and quality ensurance. It's just hard to face if you're the only code contributor and that's why I'd wished for multiple times to make my life easier which didn't happen, wasn't possible facing the official repo's strategy that's still in place (and should be according to the project leaders). I feel a little bit lost, honestly, as I understand their point of view as well as mine. But my time is limited, let's say, reconsidering: Since I decided to fork it's a fast moving app repo in here and also think even if reviews were missing, a certain (unlabeled) code quality is also there as I test very carefully before releasing stuff.

(...)
To answer your second question: If you'd like to "easy update" when a new version is available from the fork, I'd suggest you install and update via F-Droid.

**Comment by schmendrik** [comment]
@Catfriend1 thank you for explaining the situation. 

It's true, the official syncthing-android app has been slow going to say the least. Whoever is in charge of it, I feel like their update cycle is bad they should feel bad. It rendered my syncthing app entirely useless on my phone using Android 8.0. So yeah, I think I'l try the F-Droid option from now on.

**Comment by noodle-** [comment]
@Catfriend1 I installed your fork through F-Droid, not sure whether the issue is completely gone yet. There was some suspicious CPU usage before rebooting, but after that, and testing again it's looking good! 

I just quickly tested the basics, I'll just put it through normal use the following day(s) and report back. :)

**Comment by noodle-** [comment]
![cpu](https://user-images.githubusercontent.com/812565/48123952-77966580-e27b-11e8-813a-99223ff220d1.png)

So this morning my phone was not charging (anymore) and thus syncthing disabled, but there was still constant CPU usage for the next few hours (about 3?) till I restarted my phone again.

After restarting my phone I enabled syncthing to start without being charged and disabled syncthing again and now there's no constant CPU usage. So I can't pinpoint exactly when the constant CPU usage is caused/triggered/whatever.

Maybe random sidenote?: I noticed there's some network activity of about 15KB/s every 5 seconds while syncthing is running. Which is weird in itself, but at least this does not occur when syncthing is disabled and still eating CPU.

If you know any way of getting more accurate data to you compared to this guesswork, please let me know which logs might be useful.

**Comment by Catfriend1** [comment]
Hi @noodle-
Thanks for your feedback on this. Battery usage is hard to analyse but I'll do my best contacting you when I have a new idea what it could be.  Which Android device and version are you running? If you like to help me Analyse this deeper, I'd suggest rebooting the phone with the official syncthing option "start automatically on boot" disabled. To narrow it down, consider turning off the four Discovery options during log grab session , they're found in syncthing options. Global Discovery costs battery and mobile data as it holds a permanent connection to the syncthing Discovery cloud servers.
Start syncthing - pause all folders on the UI - and let it run for hours on battery. Before whilst and after exiting syncthing , capture log outputs e.g. using the matloglibre app (fdroid) . Hint: you'd need to save the log down to sdcard regularly as it's a ring buffer quickly losing contents of it is full. I would appreciate getting the detailed logs - if you like via private data transfer and some notes where you describe the main milestones of this test, for example "10.53 started app" "10.55 paused folders" "10.57 put screen off for 30 minutes" "11.27 exited syncthing app".

**Comment by Catfriend1** [comment]
Hi @jsmith037982374
I saw your post regarding the keyboard popup. Can you please try and report back if the problem is solved in the current release version of the fork? I fixed this some days ago.
Kind regards
Catfriend1

**Comment by jsmith037982374** [comment]
@Catfriend1 yes, it looks like the problem is solved in the latest version of the fork! I am new to using syncthing, but I will switch over to using your client for android; thanks for your efforts!

**Comment by Catfriend1** [comment]
Hi @o-l-a-v ,
Does the battery saving dialog bug also occur when using Syncthing-Fork?

**Comment by Catfriend1** [comment]
@NicoWeio Hi Nico, what you requested is already available in Syncthing-Fork (gplay, fdroid) where you get an explanation (and soon a green/red) when syncthing is and why it is (not) running.

**Comment by NicoWeio** [comment]
@Catfriend1 Oh, great. But will there be a PR for the "official" client?

**Comment by Catfriend1** [comment]
@NicoWeio The PR was already there, discussed back in august 2018 and rejected by the maintainer.

**Comment by Catfriend1** [comment]
@hub101 Hi, I just read your ticket on the other site and would suggest two things to get the out of sync resolved. As there is a phone involved, would you mind trying Syncthing-Fork from gplay? Secondly, please ensure that a folder shared between phone and other devices has ignore permissions set on all devices participating.

**Comment by Catfriend1** [comment]
@stuart12 Hi, Just read your ticket about crashing syncthing without indication when and why. Syncthing-Fork is optimized for Android 8+ plus can tell you if it crashed including logs. Would you mind having a try?

**Comment by stuart12** [comment]
@Catfriend1 It is too late at the moment.  I have removed the SD card (that was formatted as internal storage), reinstalled syncthing and now the errors have gone away.  I'll try syncthing-fork if similar errors return. [1267](https://github.com/syncthing/syncthing-android/issues/1267)

**Comment by Catfriend1** [comment]
@stuart12 Sure, no problem. I'm glad if it works for you now :-).

**Comment by Catfriend1** [comment]
Hi @aslmx !
Can you please check if the "syncthing doesn't start" problem also occurs in Syncthing-Fork? I'll soon make a BETA release with syncthing v1.0.1 .

**Comment by aslmx** [comment]
hi @Catfriend1  thanks for the notice. 

i can do it - not sure when exactly though... 

What do i need to do? Can i use the syncthing-fork app which is in the f-droid (v 1.0.0.20  - updated 4 days ago)? Or do i need a special release?

i guess i need to setup the sync repos again?

I'll have a look in the next days as soon as i can spare some time.

**Comment by Catfriend1** [comment]
@alaquepasa its fixed in the forked version ;-)

**Comment by Catfriend1** [comment]
@colans Hi, does the ignore device button work correctly in 1.0.1.3 app version?

**Comment by colans** [comment]
@Catfriend1 Not sure how to install it.  I found the fork on F-droid, but the latest version shows up as 0.14.52.1, which doesn't even work.  I get a 404 when I try to install.  This is the second strange thing is that newer releases are showing on https://f-droid.org/packages/com.github.catfriend1.syncthingandroid/ (even though they're not showing up in the app).  Any ideas?

**Comment by Catfriend1** [comment]
@colans Try to refresh the fdroid repository on your phone to see the current version (swipe down on main screen).

**Comment by Catfriend1** [comment]
@crocket Check Syncthing-Fork for less battery drain. It has optimizations for that and solves the issue.

**Comment by crocket** [comment]
Will the improvements of syncthing-fork be merged into syncthing?

**Comment by Catfriend1** [comment]
Currently partly or no depending on what maintainer Audrius thinks.

**Comment by crocket** [comment]
How did you solve battery problem?

**Comment by colans** [comment]
@Catfriend1 Thanks for the tip!  Your release seems to have fixed https://github.com/syncthing/syncthing-android/issues/1275 as I no longer get pings from that device.  However, that part of the UI still looks the same.  There's no Ignore button, but I suppose just backing out of the Device Add form adds it to the ignore list.  So it works, but this isn't exactly clear from the UX.

**Comment by Catfriend1** [comment]
The ignore button is in the notification...

**Comment by colans** [comment]
I didn't see one.  The only options were to swipe it away or tap it to open the Device Add form.
